### DISCUSSION TITLE
Curobo collision geometry differs from IsaacSim approximation

### DISCUSSION THREAD

**Discussion by frederiknolte** [NVlabs/curobo#495] [Software Q&A]
If it’s not a bug, please use discussions: https://github.com/NVlabs/curobo/discussions

Please provide the below information in addition to your issue:

1. cuRobo installation mode (choose from [python, isaac sim, docker python, docker isaac sim]): docker isaac sim
2. python version: 3.10
3. Isaac Sim version (if using): 2023.1.1, curobo 0.7.4

------

Hi, 
I am using custom meshes in my IsaacSim environment and when using the mesh collision checker type of curobo for motion planning, it appears as though the scene meshes themselves are used for collision detection. 
IsaacSim, on the other hand, can use various kinds of approximation for detecting collisions (convex decomposition, convex hull, etc.). Using the mesh collision detection that is offered by IsaacSim is impractical due it being extremely slow. 

The problem is now that when using shape approximations for collision detection in IsaacSim but also using curobo that relies on the meshes for collision detection, there is a mismatch regarding which positions are attainable without collisions. As a result, I often get trajectories that in principle would work but due to the collision approximation in IsaacSim the trajectories result in collisions when executed. 

Is there a way to use those collision shapes that are used in IsaacSim also for curobo motion planning and ik solving?

**Comment by balakumar-s** [comment]
cuRobo parses the USD stage to get obstacles. If you can find these approximate meshes in the USD stage, you can pass these paths to the stage parser here:

https://github.com/NVlabs/curobo/blob/0a50de1ba72db304195d59d9d0b1ed269696047f/examples/isaac_sim/motion_gen_reacher.py#L291
